src/openrct2/object/SmallSceneryObject.cpp

```diff
@@ -105,6 +105,10 @@ void SmallSceneryObject::DrawPreview(rct_drawpixelinfo* dpi, int32_t width, int3
         if (_legacyType.HasFlag(SMALL_SCENERY_FLAG_HAS_SECONDARY_COLOUR))
         {
             imageId = imageId.WithSecondary(COLOUR_YELLOW);
+            if (_legacyType.HasFlag(SMALL_SCENERY_FLAG_HAS_TERNARY_COLOUR))
+            {
+                imageId = imageId.WithSecondary(COLOUR_DARK_BROWN);
+            }
```

Comment:
With large scenery and walls, primary/secondary/tertiary colours can be set independently, i.e. having a secondary colour does not require having a primary one, which is sometimes used for consistent colouring settings across a set of scenery objects.

It would be nice if we did the same for small scenery. The only reason why small scenery is nested like this, IIRC, is because it did weird things with it flags. But we refactored that out anyway.